skale/transactions/result.py

```diff
@@ -93,6 +99,8 @@ def tx_failed(self) -> bool:
 
     def raise_for_status(self) -> None:
         if self.dry_run_failed():
+            if is_revert_error(self.dry_run_result):
+                raise SolidityError(self.dry_run_result['error'])
```

Comment:
I think it's better for consistency to raise excpetion derived from TransactionError